redisson/src/main/java/org/redisson/RedissonFairLock.java

```diff
@@ -93,134 +106,187 @@ protected void unsubscribe(RFuture<RedissonLockEntry> future, long threadId) {
+                        "return nil;" +
+                    "end;" +
+
+                    // the lock cannot be acquired
+                    // check if the thread is already in the queue
+                    "local timeout = redis.call('zscore', KEYS[3], ARGV[2]);" +
+                    "if timeout ~= false then " +
+                        // the real timeout is the timeout of the prior thread
+                        // in the queue, but this is approximately correct, and
+                        // avoids having to traverse the queue
+                        "return timeout - tonumber(ARGV[3]) - tonumber(ARGV[4]);" +
+                    "end;" +
+
+                    // add the thread to the queue at the end, and set its timeout in the timeout set to the timeout of
+                    // the prior thread in the queue (or the timeout of the lock if the queue is empty) plus the
+                    // threadWaitTime
+                    "local lastThreadId = redis.call('lindex', KEYS[2], -1);" +
+                    "local ttl;" +
+                    "if lastThreadId ~= false and lastThreadId ~= ARGV[2] then " +
+                        "ttl = tonumber(redis.call('zscore', KEYS[3], lastThreadId)) - tonumber(ARGV[4]);" +
+                    "else " +
+                        "ttl = redis.call('pttl', KEYS[1]);" +
+                    "end;" +
+                    "local timeout = ttl + tonumber(ARGV[3]) + tonumber(ARGV[4]);" +
+                    "if redis.call('zadd', KEYS[3], timeout, ARGV[2]) == 1 then " +
+                        "redis.call('rpush', KEYS[2], ARGV[2]);" +
+                    "end;" +
+                    "return ttl;",
+                    Arrays.<Object>asList(getName(), threadsQueueName, timeoutSetName),
+                    internalLockLeaseTime, getLockName(threadId), threadWaitTime, currentTime);
         }
-        
+
         throw new IllegalArgumentException();
     }
-    
+
     @Override
     protected RFuture<Boolean> unlockInnerAsync(long threadId) {
         return commandExecutor.evalWriteAsync(getName(), LongCodec.INSTANCE, RedisCommands.EVAL_BOOLEAN,
                 // remove stale threads
-                "while true do "
-                + "local firstThreadId2 = redis.call('lindex', KEYS[2], 0);"
-                + "if firstThreadId2 == false then "
-                    + "break;"
-                + "end; "
-                + "local timeout = tonumber(redis.call('zscore', KEYS[3], firstThreadId2));"
-                + "if timeout <= tonumber(ARGV[4]) then "
-                    + "redis.call('zrem', KEYS[3], firstThreadId2); "
-                    + "redis.call('lpop', KEYS[2]); "
-                + "else "
-                    + "break;"
-                + "end; "
-              + "end;"
-                
-              + "if (redis.call('exists', KEYS[1]) == 0) then " + 
-                    "local nextThreadId = redis.call('lindex', KEYS[2], 0); " + 
+                "while true do " +
```

Comment:
Please don't include non-changed code

Reply:
I'm not sure what you're asking for here? I've reformatted the code (to make it consistent across the file and easier to read) and added comments.

Reply:
Since the logic of code hasn't been changed please keep it untoched.